### ISSUE TITLE
[FEATURE] Support for Leica Wild T1000

### ISSUE THREAD

**Issue by carlopav** [steko/totalopenstation#167]
Hello, thanks for this great initiative. I have an old Wild T1000+Distomat total station. After some attempts with TOPS to download the data straight to the pc via the connection cable and a serial to usb adapter, that leaded to a code 22 error, I succesfully download the data with meridiana software using this settings: Baudrate 2400 parity EVEN bitesize 7 stopbit 2 (and input SET, 5, 9, 8, RUN in the total station consolle).
Any hint would be awesome. Thanks!

**Comment by psolyca** [comment]
Have you tried the custom settings with the same values as Meridian software ?
We know we have trouble with serial link, see https://github.com/totalopenstation/totalopenstation/pull/165.
We are trying to hande this.

**Comment by carlopav** [comment]
> Have you tried the custom settings with the same values as Meridian software ?

Yes, I did, but I think there are some problems with the serial connection, as you mentioned. Let me know if I can help testing some solution.

**Comment by carlopav** [comment]
Just a small reminder to me, in case i'll have time to dig into the code. This is an excerpt of the T1000 user manual:

![immagine](https://user-images.githubusercontent.com/47068848/183512895-07d8b7fa-2ba7-4c14-95c2-ec8ff20ec2c8.png)

**Comment by carlopav** [comment]
Anyone available next 10days for a screenshare to help me to connect to the T1000? I'll be happy to contribute if i'm able to use tops at work :)

**Comment by steko** [comment]
> Anyone available next 10days for a screenshare to help me to connect to the T1000? I'll be happy to contribute if i'm able to use tops at work :)

I'm happy to do that. Do you have any specific day and time that is best? What about Tuesday 20 in the morning?

**Comment by carlopav** [comment]
Thanks so much! I really appreciate. What about wednesday/thursday/Friday/saturday morning? Unluckly in not available next thursday :(

**Comment by enzococca** [comment]
Hi,

I'm looking into adding support for Leica Wild T1000. To implement this, I would need:

1. **Sample data file** - A text file exported from or downloaded from the device
2. **Communication protocol details** - Baud rate, parity, etc. (you mentioned: 2400 baud, EVEN parity, 7 data bits, 2 stop bits)
3. **Data format documentation** - Or at least a description of the data format

From your description, it seems like the connection settings are different from standard Leica instruments. If you could provide a sample data file, I can work on creating a parser for this format.

Thanks!

**Comment by carlopav** [comment]
Ciao @enzococca,
I'm sorry, I didn't see your message. I asked claude to debug, and after some try and error it ended up with a working solution.
The transfer now works flawlessly.
Just is quite slow and it stucks tops for 20 sec to download 28 lines :) The no visual feedback could lead the user to think it's not working. But it is :)
Thanks also @steko for the great availability in screensharing to try to hand implement it.
